### DISCUSSION TITLE
i915 issue

### DISCUSSION THREAD

**Discussion by decodering22** [PeterSuh-Q3/tinycore-redpill#29] [Q&A] [answered]
After boot kernel panic and can't install DSM due to i915 issue. Is the i915 built into this latest build?

**Comment by PeterSuh-Q3** [comment]
SA6400 ?

**Comment by decodering22** [comment]
Hi Peter yes SA6400, been pulling my hair trying to get xpenology installed and no luck

**Comment by PeterSuh-Q3** [comment]
BareMetal or VM like Proxmox?
And what is the CPU model?

**Comment by decodering22** [comment]
Baremetal intel i5-12500 on MSI MGP 760I motherboard. If there is anything you want me to try let me know I'm in front of it now, been battling this for a few hours

**Comment by PeterSuh-Q3** [comment]
There is no space in the message box of the Henology forum. I wish we would only talk on GitHub.

![스크린샷 2024-02-06 오후 12 39 05](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/85427533/42914c45-8f94-4f43-9cf8-401cabbaff13)

Are all ARC/RR/mshell in the same situation?

https://xpenology.com/forum/topic/60130-redpill-tinycore-loader-installation-guide-for-dsm-71-baremetal/?do=findComment&comment=456343

**Comment by decodering22** [comment]
I only tried ARC and RR. Initialy i tried RR and it just kernel panic so assumption is i915 is the issue but I'm not sure. I tried your RR build and after the QR code pops up then screen goes blank. I assumed based on that thread the latest i915 was loaded on i915 so it should work. Not sure what it could be b/c there is no DSM log since I can't even install DSM. 

do you want to email or even call me?

**Comment by PeterSuh-Q3** [comment]
https://xpenology.com/forum/topic/60130-redpill-tinycore-loader-installation-guide-for-dsm-71-baremetal/?do=findComment&comment=456343

Looking at this log, it appears that mshell has already been tried.

Are you sure that i915 is causing the panic by checking the logs?
It's hard to be sure until you check dmesg yourself.
rr and mshell provide functions to retrieve and analyze dmesg from error situations.

The fact that the screen disappears on SA6400 may mean that i915.ko is working well.
The bug where the screen disappears after i915.ko is loaded properly is also confirmed in mshell.

It seems to me that the Realtek 2.5G network is not operating stably.
Are there any additional NICs that can be installed? It's better if it's from Intel.

**Comment by decodering22** [comment]
I have a 10Gbe intel card i plan to use you think if i disable the onboard Realtek that will fix the issue? I can disable it and test it for sure that is not a problem i will disable it on the UEFI bios. 

I can't check the log b/c as soon as the IP loads it causes the kernel panic. The only reason i went down the i915 rabbit hole is b/c it was mentioned that was the issue. Let me pop in the intel X520-DAC1 and disable the Realtek 2.5G to see if that works or not. 

Where can i pull the log to be 100% sure?

**Comment by decodering22** [comment]
I also have a 1GBe intel card i will pop in for the test as well

**Comment by decodering22** [comment]
Same thing, i disabled the Realtek 2.5Gbe NIC in the BIOS itself, it detected the new pcie 1Gbe Intel VRRH1-HIGH Nic without issues. and the result is the same. Screen goes black and after about 15 seconds I can ping it again but can't be found in finds.synology.com or the IP with port to connect to it. 

Can you tell me where to pull the log from and when. Really appreciate your help

**Comment by decodering22** [comment]
If there is some settings you want me to try and test let me know. Not sure how to pull the log b/c it kernel panics right away

**Comment by PeterSuh-Q3** [comment]
In the case of mshell, the last log file is saved when the Friend kernel is loaded.
It exists in the directory below.
/mnt/tcrp-p1/logs/jr
Friend, stop with ctrl+c while booting.
Please examine the files under this directory.
You should also look at dmesg, but the contents of linuxrc.syno.log are also important.
Can you provide me with these 2 files for download?

![스크린샷 2024-02-06 오후 1 17 30](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/85427533/50c9b453-c44a-4361-8b43-06faa6007a55)

**Comment by decodering22** [comment]
Yes let me load RR on a flash drive right now and get these to you now. Give me a min I have to go through the entire process probably a few mins to get it to the point where it does this. Do you want me to email it to you directly or do you want to continue here on github?

**Comment by PeterSuh-Q3** [comment]
I gave an example of MSHELL. I don't know exactly where and how RR's logs are stored.
GitHub is also good enough.

**Comment by decodering22** [comment]
I have your tinycore-redpill that is what I meant when i said RR not sure if your reference to RR is something else but i assume the redpill is the MSHELL which is what I loaded prior and now

**Comment by decodering22** [comment]
I'm following this procedure although at the end it doesn't exactly work per the instructions
https://xpenology.club/install-dsm-7-on-baremetal-or-vm/

**Comment by decodering22** [comment]
Ok at step Build the Loader 7.2.1.-69057, Friend) and about to reboot

**Comment by decodering22** [comment]
Ok just rebooted 
![Uploading 20240205_203814.jpg…]()

**Comment by decodering22** [comment]
Ok just went to blank screen, I'm going to pull the logs like you said and post it
![20240205_203955](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/78891534/6c15abce-d46d-481c-81f9-42915107789c)

**Comment by decodering22** [comment]
Peter there is no log in the /mnt/tcrp-p1# folder

**Comment by decodering22** [comment]
what is the user and pass default so i can use WinSCP to transfer it over and attach it here?

**Comment by decodering22** [comment]
I can keep trying ot make it crash to make it have a log but its not storing a log. 

![20240205_204713](https://github.com/PeterSuh-Q3/tinycore-redpill/assets/78891534/003c7c7d-2b0c-4851-be6d-937847529e04)

**Comment by decodering22** [comment]
Need to figure out how to export the dmesg so i can copy it over. Was hoping to use WinSCP but i don't know the user and password as its not tc & P@ssw0rd anymore

**Comment by decodering22** [comment]
Peter how do i get the dmesg i saved to a txt file off so i can transfer it over to windows to send it to you?

**Comment by decodering22** [comment]
Ok i was able to connect using the PC and ran the dmesg and copied and paste from the Browser. Let me know what else I can provide. I don't see the logs in the other foler?

[dmesg.txt](https://github.com/PeterSuh-Q3/tinycore-redpill/files/14174973/dmesg.txt)

**Comment by decodering22** [comment]
shouldn't there be some reference to i915 in dmesg?

**Comment by decodering22** [comment]
Any idea why it is not logging in /mnt/tcrp-p1/  ??? I can make it repeat go to black screen but it doesn't log in this area

**Comment by decodering22** [comment]
Anything else you want me to grab?

**Comment by decodering22** [comment]
Another note after the screen goes blank it takes down the NIC with it completely likely due to kernel panic for what its worth. 

Let me know your thoughts after looking at the dmesg output. there is no linuxrc.syno.log b/c DSM install never got started b/c of the kernel panic

**Comment by decodering22** [comment]
<img width="604" alt="Capture" src="https://github.com/PeterSuh-Q3/tinycore-redpill/assets/78891534/78890ebb-40b1-411b-96a2-925665b43b7f">

**Comment by decodering22** [comment]
Calling it a night Peter tomorrow if you have any idea let me know. If there is any specific BIOS settings you think could be it let me know. I'm willing to try everything, spend a good bit of money on this and if it doesn't work its all for nothing. thanks for your help

**Comment by PeterSuh-Q3** [comment]
I'm sorry. I didn't have time to check GitHub because I was busy with work.

I don't know if you understand this script, but
This is the core content that contains the log.
I don't think It had a chance to copy the logs from the two events - the earliest and the latest when Junior boots.

There are two more events in between. (There are two types: jrExit and modules.)
I will improve the dbgutils Addon so that it can retrieve logs from these two events.

I will share this addon again when it is more complete.

https://github.com/PeterSuh-Q3/tcrp-addons/blob/main/dbgutils/src/install.sh

**Comment by decodering22** [comment]
Hi Peter, do you want me to manually run this script? Just trying to understand what I can do now to help isolate the problem and get this fixed?

**Comment by decodering22** [comment]
Did you see the dmseg output i uploaded? Anything else I can do now to fix this

**Comment by PeterSuh-Q3** [comment]
This page has already been changed and you cannot do this yourself. Logging out is one of the most difficult improvement tasks. I have already tried this several times and failed. I'll try it one more time and after a few days, I hope you don't rush and give me time.

**Comment by decodering22** [comment]
Hi Peter, I have a limited time before I must return the hardware for a refund. This is why I'm trying desperately now to get everything working. A few days is fine but is there anything else i can do in the meantime to help narrow the problem? Anything at all i can try? BIOS settings, or any other option to ensure it is not i915?

**Comment by PeterSuh-Q3** [comment]
The most stable and efficient way to view logs is to monitor console output through the serial com port. However, since this device is no longer supported in the new MOBO, we did not mention this from the beginning. It is also easy to check for kernel panic or NIC unresponsiveness here. There is something not mentioned about how to check IP. The IP displayed on the FRIEND screen may not be 100% reliable. So as a workaround, I use finds.synology.com and if it doesn't show up, I look into the IP assigned to my router. The best way is to fix the IP by pre-setting the Mac address you know to the router. Try securing an IP address using this method first.

**Comment by decodering22** [comment]
Hi Peter, i have both finds.synology.com up in a browser and I'm pinging and using the browser to connect, all 3 at the same time. I can confirm when the "kernel panic" occurs it drops the IP all together and the finds.synology.com i'm refresing the whole time never finds the DSM. You are right there is no more serial port. Is there any other way while you work on that script i can try different settnigs or options you can think of?

**Comment by PeterSuh-Q3** [comment]
Although there are not many cases, there has been no report of i915 kernel panic when using SA6400 bare metal on a 12th generation CPU.

**Comment by decodering22** [comment]
That is probably true, so what else could it be?

**Comment by PeterSuh-Q3** [comment]
Analysis of logs is currently the best method; anything else can only be guesswork. You keep mentioning kernel panics, but even in non-kernel panic situations, the loader can crash with an error. And it seems that any hardware option adjustments will no longer be effective. Redfill models that can operate in the 12th generation are limited, and the SA6400 is a model that has just entered stabilization, so more users' experience is needed. Other than my Mshell, another option may be the rr loader from a developer called wjz304. I know that the sharing of this loader has been closed, but if there is a way to do so, I would like to obtain it and try it.

**Comment by decodering22** [comment]
Sure what ever you want me to try i will do it, just let me know. The sooner you can get me the script to look at the logs the better. Just let me know what you want me to try. Are you are correct it could be other issue causing it. Just need to narrow it down

**Comment by decodering22** [comment]
Peter was the dmseg output of any value when you looked at it?

**Comment by decodering22** [comment]
I just tried this ARPL i think it is by wjz304, ti does the same it hangs. but I don't think there is anyway to pull any logs from it?
https://github.com/fbelavenuto/arpl/releases/tag/v1.1-beta2a

**Comment by decodering22** [comment]
Peter I also want to mention, i have no problem paying you for your time helping me. I can paypal you just let me know. Thanks again

**Comment by decodering22** [comment]
you are correct i can't get access to this repo https://github.com/wjz304/arpl-i18n/releases 

If you have it i can try it

**Comment by PeterSuh-Q3** [comment]
Use this
https://github.com/syno-community/rr/releases

**Comment by decodering22** [comment]
Thanks Peter i will try this as well. I ordered a new motherboard as well and it should be here today. I took the other system apart already so when the new one arrives I will try this on it.

**Comment by PeterSuh-Q3** [comment]
These are successful cases using Intel 12th generation and SA6400, including N100, at the server forum in Korea where I work. It would have been better if you had let me know in advance.
good luck.

https://svrforum.com/?act=&vid=&mid=case&category=&extra_vars5=12&extra_vars6=&extra_vars7=&extra_vars8=&extra_vars9=&extra_vars10=

**Comment by decodering22** [comment]
Thanks Peter I'll translate the site and see. It has to be some compatibility with the Motherborad i was using. I'm still waiting for the new one to arrive, it does not have any Realtek NIC, only Intel, but still not sure if that was even the issue since we couldn't capture logs before it crashes. 

I'll let you know as soon as I have the new system to see if it works.

**Comment by decodering22** [comment]
Hi Peter, so I had some luck with this new motherboard and it is working but an odd issue is if i remove the USB drive it just goes straight to the BIOS. It seems like i have to have the USB drive plugged in all the time so it forces DSM to boot. Is this normal? Is there a way to make the SATA drive bootable so no USB is needed?

**Comment by decodering22** [comment]
Peter just so i understand the loader needs a USB drive to boot, than it installs the DSM on the SATA SSD correct. Is it possible to partition part of the SATA SSD so it is used to boot the DSM and act like the loader?

**Comment by PeterSuh-Q3** [comment]
It's impossible. You need to understand the mechanism by which Synology disks' boot devices and data disks work. All data disks have the same system partition at the front of the disk, and DSM is installed equally there. DSM itself does not support booting. Genuine devices required for Synology booting use durable Sata Dom. Redpill loaders such as TCRP and ARPL create boot loaders with the same structure as this sata dom. Small capacity USB, MSata, SSD, etc. must be made independent with a separate bootloader. M.2 Sata SSD is also available, but Nvme is not available.

**Comment by decodering22** [comment]
If i understand the USB or SATA DOM is needed no matter what. The SATA disk and M.2 based on what you stated both have the same DSM system partition. So if I add a second M.2 drive its not just a Volume drive only, but there is a small system partition as well correct?  I'm going to just use a small low profile and fast USB and call it a day. I was hoping to avoid it but it doesn't sound like it is possible. 

It is only used during boot only correct, so minimal Read/Write on the USB? I don't reboot often so I suspect it should be fine?

**Comment by PeterSuh-Q3** [comment]
There are two types of M.2, so they must be clearly distinguished.
These are M.2 Sata SSD and M.2 NVMe SSD.
If you add an M.2 Sata SSD, it operates the same as a regular SATA (SSD/HDD) disk.
Any Sata based disk can be used as a bootloader.
However, M.2 NVMe SSD can only exist as a cache, but with the help of a special addon, it can be turned into a data disk volume. It cannot be used as a bootloader.

USB sticks are the most vulnerable devices in terms of durability.
It stays on the PC during booting and does not participate in reading/writing, but
Must be exposed to continuous heat.
This is a factor that reduces the durability of the USB stick.

**Comment by decodering22** [comment]
Hi Peter, great explanataion, I know the difference between the two types of M.2 drives and how they are utilized in xpenology pradigm with m.2 as cache but with the drive DB update being able to be used as data disk volume. 

Tracking that M.2 NVMe PCIe can't be used for bootloader and right now it isn't fully supported without a SATA drive. 

What I was concerned with was the USB drive being suseptiple to failure and if it was possible to partition the SATA SSD drive for a boot partition and the other partition being used for the DSM System partition. It sounds like that is not possible at the moment and the use of a USB or separate SSD(SATA) or SATA DOM is needed for the boot loader alone. 

I was under the impression with a previous version (Jun's) release in DSM 6.x days he had figured out a way to use the SSD (SATA) as both a bootloader and DSM system partition too. I just assumed that was also possible now with DSM 7.2 and beyond. 

In your opinion is it worth it to transition to a SATA DOM for long term reliability over USB?

**Comment by PeterSuh-Q3** [comment]
I've seen users using SATA Doms instead of actual USB disks. I haven't directly encountered verified data on durability, but isn't the fact that Synology is using it enough verification?

**Comment by decodering22** [comment]
Yea if Synology is using a SATA DOM than it must be reliable enough. I don't mind getting a small cheap SATA drive. I'm just concerned the USB disk failing overtime like you mentioned due to heat. The only options for SATA DOM are supermicro i have found and they don't have the greatest review. I have plenty of SSD lying around and there is plenty of open SATA slots. For now I ordered a Sandisk Fit and will use that until it fails than transfer to another option. Thanks for your help

**Comment by PeterSuh-Q3** [comment]
/var/log/linuxrc.syno.log

The file shows most of what REDPILL does.
However, the clean log in SA6400 is due to the characteristics of the EUDEV method.
In the EUDEV method, most important parts of the log are complex enough to be judged by dmesg.
As you can see, dmesg contains quite a bit of content.

Whether differences in BIOS versions really cause kernel panics or other errors can be determined only through logs.

To facilitate collection of this log, the NIC must be operating at a minimum level.

The most serious problem is when even the NIC does not work, and it becomes difficult to determine the cause.

If only the NIC is working, use a web browser
IP address: 7681 via port connection
You can check the causes of the problem using /var/log/linuxrc.syno.log or dmesg.

An alternative for when the NIC is not working is console monitoring through the Serial COM1 port.
Since the Serial COM1 port is no longer supported by the latest MOBO, it is currently viewed as an unusable solution.

I am using the dbgutils addon that is used with rr/arc/mshell to easily check for these problems.

This is a function that stores various logs, including the two files mentioned above, which contain the situation up to a possible kernel panic during mshell booting, in a specific location and allows them to be checked separately.
It is not yet known whether this function works reliably.

The best thing is console monitoring through the Serial COM1 port.
For this purpose, I deliberately purchased an interface card with two physical serial ports.
However, the ADDRESS mapping that allows COM1 port to operate is not possible, so I have not yet been able to get it to work properly.

I'm still looking for an interface card that physically allows ADDRESS adjustment with DIP switches.

Efforts to monitor logs are still ongoing, but a stable solution has not yet been found.

**Reply by decodering22** [reply]
Hi Peter, even though I can ping and receive replies i cna't connect via IP:7681, no connection of any kind is working after the bootloader finishes loading and screen goes blank. 

I don't even know what could possibly be in the BIOS causing this. It really makes no sense but given the chipset is the same and same hardware between the boards it must be a setting in the BIOS likely.

**Comment by PeterSuh-Q3** [comment, marked as answer]
And, i226-v is still a chipset with many issues.
There have been several igc module patches for this in the past, but
There are no more patches in progress recently.
If you do a Google search on i226-v and EEE (Energy Efficient Ethernet) issues, you will get an idea of the situation.

So I also bought a NIC for this chipset.
Although this is the case with mshell, it is case by case.
This NIC works in one MOBO but not in another MOBO.
I don't know why.
So, I would still recommend avoiding the i226-v chipset if possible.
However, the latest MOBO uses a lot of i226-v.

**Reply by decodering22** [reply]
That is interesting, but your deducing the problem is i226-v, if you remember initially i was chasing i915. Not saying the NIC can't be the problem but the MB which works has the same exact i226-v NIC. The funny thing is it also has a second 1Gbe NIC and that doesn't work, what works is the i226-v 2.5G port. 

I guess the mystery of why it works in one MB vs another will remain and could be the culprit instead of something in the BIOS itself. Chris thinks the problem is the BIOS, he also thought it was i915 initially too. 

I just wish i could nail down the problem. I'm sending the new MSI board back and sticking with the ASrock for now since it works.
